plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/TableSnapshot.java

```diff
@@ -300,7 +301,8 @@ private Stream<DeltaLakeTransactionLogEntry> getV2CheckpointTransactionLogEntrie
     {
         return getV2CheckpointEntries(session, entryTypes, metadataEntry, protocolEntry, checkpointSchemaManager, typeManager, stats, checkpoint, checkpointFile, partitionConstraint, addStatsMinMaxColumnFilter, fileSystem, fileSize)
                 .mapMulti((entry, builder) -> {
-                    if (entry.getSidecar() == null) {
+                    // Sidecar files contains only ADD and REMOVE entry types. https://github.com/delta-io/delta/blob/master/PROTOCOL.md#v2-spec
+                    if (entry.getSidecar() == null || !isDataRequested(entryTypes)) {
```

Comment:
nit: We could exclude unnecessary types from `entryTypes` before passing to `CheckpointEntryIterator` constructor. Probably, there's no such method usage though.